src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -117,9 +120,11 @@ public partial class AuthenticationResult
             IAuthenticationScheme authenticationScheme,
             Guid correlationID,
             TokenSource tokenSource, 
-            ApiEvent apiEvent)
+            ApiEvent apiEvent,
+            ICacheSessionManager cacheManager)
         {
```

Comment:
Looking at the spec, it looks like AuthenticationResult should include `.GetIdTokenClaims()` which would return just the IdToken Claims. I think id_token claims and tenant profiles are not related features, so a customer might want the claims but not necessary the tenant profile.

Reply:
This is not a part of this issue. As discussed earlier, once this is done we will decide how to expose the claims. @jmprieur to comment

Reply:
I created a small extension method that computes `GetIdTokenClaims()`:

```CSharp
 public static class AuthenticationResultExtensions
    {
        public static IEnumerable<Claim> GetIdTokenClaims(this AuthenticationResult authenticationResult)
        {
            return authenticationResult.Account.TenantProfiles[authenticationResult.TenantId].Claims.Claims;
        }
    }
```

It works most of the times, but it does not work if we use `AcquireTokenSilent` with a `.WithAuthority()` calls which is called with a tenant expressed as a **domain** instead of the GUID (which is perfectly legit). Indeed, the tenantID in the tenant profile will be the domain name, not the tenant ID, whereas the authenticationResult.TenantId is the GUID

```CSharp
  result = await app.AcquireTokenSilent(new[] { "user.read" }, result.Account)
                .WithAuthority(app.Authority.Replace("common", "msidentitysamplestesting.onmicrosoft.com"))
                .ExecuteAsync();
```

We might have a little bug here:
cc: @jennyf19 @neha-bhargava

Reply:
Was thinking that ClaimsPrincipal has everything, is there a need for IEnumerablem<Claim> ?